### PR TITLE
Change find_interlinks return type to list of tuples

### LINKED ISSUE #2635: Wikicorpus interlinks storage should be list of tuples instead of dict.
#### Problem description

I am trying to get the list of interlinks occurring on a wikipedia page. The current parser uses a dict to store the links. However, the same link might be referred to using different text on the same page. 

E.g. for the `Anarchism` page in the test script the `Anarchism in Italy` link is referred using 2 different texts. 

```
Anarchism in Italy -> Italian anarchists
Anarchism in Italy -> Italy
Anarchism in Italy -> Italy
```

This motivates the reasoning that interlinks should be stored as a list of tuples instead of a dictionary. 

#### Steps/code/corpus to reproduce

Include full tracebacks, logs and datasets if necessary. Please keep the examples minimal ("minimal reproducible example").

The issue is at the following line: 
https://github.com/RaRe-Technologies/gensim/blob/3e027c252eac3cf7e613f425ad8b070e8fe88065/gensim/corpora/wikicorpus.py#L174

And the subsequence assignment in later lines. 
The above code only keeps track of the last interlink mapping instead of all interlink mappings in the text. 

I will send a PR resolving this issue. 

#### Versions

Please provide the output of:

```python
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("NumPy", numpy.__version__)
import scipy; print("SciPy", scipy.__version__)
import gensim; print("gensim", gensim.__version__)
from gensim.models import word2vec;print("FAST_VERSION", word2vec.FAST_VERSION)
```

Output of above: 

```
Darwin-18.7.0-x86_64-i386-64bit
Python 3.7.4 (default, Aug 13 2019, 15:17:50) 
[Clang 4.0.1 (tags/RELEASE_401/final)]
NumPy 1.17.2
SciPy 1.3.1
gensim 3.8.1
FAST_VERSION -1
```

### PR DESCRIPTION
**Pull request by napsternxg**
Fix RaRe-Technologies/gensim#2635

This commit changes the interlinks storage in the `segment_wiki` script from dictionary to a list of tuples.

We can process the test wikidata used in the test suite of gensim to inspect the new behavior.
```
python gensim/scripts/segment_wiki.py -i \
    -f ~/Downloads/enwiki-latest-pages-articles1.xml-p000000010p000030302-shortened.bz2 \
    -o ~/Downloads/enwiki-latest.json.gz
```

We get the following output:

```
$ cat ~/Downloads/enwiki-latest.json.gz | zcat | head -1 | jq -r '.interlinks[] | [.[0], .[1]] | @tsv' | sort | head
-ism	-ism
1848 Revolution	1848 Revolution
1917 October Revolution	1917 October Revolution
6 February 1934 crisis	February 1934 riots
A. S. Neill	A. S. Neill
AK Press	AK Press
Abu Hanifa	Abu Hanifa
Adolf Brand	Adolf Brand
Adolf Brand	Adolf Brand
Adolf Hitler	Hitler
```

All tests pass for the related test file.

```
python -m unittest gensim.test.test_scripts
/Users/smishra/miniconda3/envs/TwitterNER/lib/python3.7/bz2.py:131: ResourceWarning: unclosed file <_io.BufferedReader name='/Users/smishra/workspace/codes/python/gensim/gensim/test/test_data/enwiki-latest-pages-articles1.xml-p000000010p000030302-shortened.bz2'>
  self._buffer = None
ResourceWarning: Enable tracemalloc to get the object allocation traceback
.....
----------------------------------------------------------------------
Ran 5 tests in 6.298s

OK
```

### CODE REVIEW

File: gensim/test/test_scripts.py [hunk 1]
```diff
@@ -105,9 +106,11 @@ def test_segment_and_write_all_articles(self):
         self.assertEqual(section_titles, self.expected_section_titles)
 
         # Check interlinks
-        self.assertTrue(interlinks['self-governance'] == 'self-governed')
-        self.assertTrue(interlinks['Hierarchy'] == 'hierarchical')
-        self.assertTrue(interlinks['Pierre-Joseph Proudhon'] == 'Proudhon')
+        # NOTE: after loading the interlinks will be loaded as list of lists instead of list of tuples
```
Comment (mpenkov) [inline gensim/test/test_scripts.py]: What's the point of this note?

Comment (napsternxg) [reply gensim/test/test_scripts.py]: This is there to ensure while writing tests after loading the json file, one does not try to compare against a tuple, as json always stores things as lists.

Comment (mpenkov) [reply gensim/test/test_scripts.py]: I think it'd be more helpful if you include your response above in the message.

The current note kind of looks like it's stating the obvious, and doesn't add any value to the code.

Comment (napsternxg) [reply gensim/test/test_scripts.py]: Just to clarify, are you suggesting that I change the comment to this: 

```python
# NOTE: To ensure while writing tests which load the generated json file, we need to ensure that the comparison of interlinks value is not done against a tuple, as json always stores things as lists. After loading, the interlinks will be loaded as list of lists instead of list of tuples. 
```

File: gensim/test/test_scripts.py [hunk 2]
```diff
@@ -70,9 +70,10 @@ def test_segment_all_articles(self):
         self.assertTrue(first_sentence in first_section_text)
 
         # Check interlinks
-        self.assertTrue(interlinks['self-governance'] == 'self-governed')
-        self.assertTrue(interlinks['Hierarchy'] == 'hierarchical')
-        self.assertTrue(interlinks['Pierre-Joseph Proudhon'] == 'Proudhon')
+        self.assertEqual(len(interlinks), 685)
+        self.assertTrue(interlinks[0] == ("political philosophy", "political philosophy"))
```
[the remark this answers is not in the document]
Comment (napsternxg) [reply gensim/test/test_scripts.py]: Done

File: gensim/test/test_scripts.py (hunk 1 above)
Comment (mpenkov) [reply gensim/test/test_scripts.py]: Ugh, that's a bit too wordy and indirect. How about something shorter and more direct?

```python
# JSON has no tuples, only lists. So, we convert lists to tuples explicitly before comparison.
self.assertEqual(tuple(blah_list), foo_tuple)
```

Comment (napsternxg) [reply gensim/test/test_scripts.py]: I like this much better.

Comment (napsternxg) [reply gensim/test/test_scripts.py]: @mpenkov done.

**Comment by mpenkov** [comment]
@napsternxg Merged. Thank you for taking care of this!
